tools/nightly_checkout.py

```diff
@@ -0,0 +1,252 @@
+    listing = glob.glob(os.path.join(source_dir, "*.pyd"))
+    listing.append(os.path.join(source_dir, "version.py"))
+    listing.extend(glob.glob(os.path.join(source_dir, "lib", "*.lib")))
+    listing.extend(glob.glob(os.path.join(source_dir, "lib", "*.dll")))
+    listing.append(os.path.join(source_dir, "bin"))
+    return listing
+
+
+def _get_listing(source_dir, platform):
+    if platform.startswith("linux"):
+        listing = _get_listing_linux(source_dir)
+    elif platform.startswith("osx"):
+        listing = _get_listing_osx(source_dir)
+    elif platform.startswith("win"):
+        listing = _get_listing_win(source_dir)
+    else:
+        raise RuntimeError(f"Platform {platform!r} not recognized")
+    return listing
+
+
+def _remove_existing(trg, is_dir):
+    if os.path.exists(trg):
+        if is_dir:
+            shutil.rmtree(trg)
+        else:
+            os.remove(trg)
+
+def _move_single(src, target_dir, mover, verb):
+    is_dir = os.path.isdir(src)
+    base = os.path.basename(src)
+    trg = os.path.join(target_dir, base)
+    _remove_existing(trg, is_dir)
+    # move over new files
+    if is_dir:
+        os.makedirs(trg, exist_ok=True)
+        for root, dirs, files in os.walk(src):
+            relroot = root[len(src):]
+            for name in files:
+                relname = os.path.join(relroot, name)
+                s = os.path.join(src, relname)
+                t = os.path.join(trg, relname)
+                print(f"{verb} {s} -> {t}")
+                mover(s, t)
+            for name in dirs:
+                relname = os.path.join(relroot, name)
+                os.makedirs(os.path.join(trg, relname), exist_ok=True)
+    else:
+        print(f"{verb} {src} -> {trg}")
+        mover(src, trg)
+
+
+def _copy_files(listing, target_dir):
+    for src in listing:
+        _move_single(src, target_dir, shutil.copy2, "Copying")
+
+
+def _link_files(listing, target_dir):
+    for src in listing:
+        _move_single(src, target_dir, os.link, "Linking")
+
+
+@timed("Moving nightly files into repo")
+def move_nightly_files(spdir, platform):
+    """Moves PyTorch files from temporary installed location to repo."""
+    # get file listing
+    source_dir = os.path.join(spdir, "torch")
+    listing = _get_listing(source_dir, platform)
+    target_dir = os.path.abspath("torch")
+    # copy / link files
+    if platform.startswith("win"):
+        _copy_files(listing, target_dir)
+    else:
+        _link_files(listing, target_dir)
+
+
+def install():
+    """Development install of PyTorch"""
+    deps, pytorch, platform = conda_solve()
+    deps_install(deps)
+    pytdir = pytorch_install(pytorch)
+    spdir = _site_packages(pytdir, platform)
+    checkout_nightly_version(spdir)
+    move_nightly_files(spdir, platform)
+    pytdir.cleanup()
+    print("-------\nPyTorch Development Environment set up!\nPlease activate to "
+          "enable this environment:\n  $ conda activate pytorch-deps")
```

Comment:
Maybe go ahead and just run the activate command at the end of the script?

Reply:
I don't think that is possible.  The activation needs to happen in the calling shell session.  If you called it in the script here, it would disappear as soon as the process ended.